### PR TITLE
Allow a docker image to be used instead of building nimbus_beacon_node

### PR DESCRIPTION
**Pull request by zah**
This would be useful when testing nimbus-eth1 in merge-mode simulations

### CODE REVIEW

File: scripts/launch_local_testnet.sh [hunk 1]
```diff
@@ -151,6 +153,14 @@ while true; do
       KILL_OLD_PROCESSES="1"
       shift
       ;;
+    --eth2-docker-image)
+      ETH2_DOCKER_IMAGE="$2"
+      shift 2
+      # TODO The validator client is still not being shipped with
+      #      our docker images, so we must disable it:
```
Comment (stefantalpalaru) [inline scripts/launch_local_testnet.sh:160]: When you want to add it, here's where you do it: https://github.com/status-im/nimbus-eth2/blob/5e9625c1be57045f23c8739c727c6257d0ecddac/docker/dist/entry_point.sh#L20

File: scripts/launch_local_testnet.sh [hunk 2]
```diff
@@ -223,11 +228,25 @@ if [[ "${HAVE_LSOF}" == "1" ]]; then
...
 if [[ "$ENABLE_LOGTRACE" == "1" ]]; then
   BINARIES="${BINARIES} logtrace"
 fi
 
+if [[ ! -z "$ETH2_DOCKER_IMAGE" ]]; then
+  DATA_DIR_FULL_PATH=$(cd "${DATA_DIR}"; pwd)
+  # CONTAINER_DATA_DIR must be used everywhere where paths are supplied to BEACON_NODE_COMMAND executions.
+  # We'll use the CONTAINER_ prefix throughout the file to indicate such paths.
+  CONTAINER_DATA_DIR=/home/user/nimbus-eth2/testnet
+  BEACON_NODE_COMMAND="docker run -v /etc/passwd:/etc/passwd -u $(id -u):$(id -g) --net=host -v ${DATA_DIR_FULL_PATH}:${CONTAINER_DATA_DIR}:rw $ETH2_DOCKER_IMAGE"
```
Comment (stefantalpalaru) [inline scripts/launch_local_testnet.sh:242]: I think ":rw" is already the default.

File: scripts/launch_local_testnet.sh [hunk 3]
```diff
@@ -245,6 +264,10 @@ cleanup() {
   for BINARY in ${BINARIES}; do
     rm build/${BINARY}
   done
+
+  if [[ ! -z "$ETH2_DOCKER_IMAGE" ]]; then
+    docker rm $(docker stop $(docker ps -a -q --filter ancestor=$ETH2_DOCKER_IMAGE --format="{{.ID}}"))
```
Comment (stefantalpalaru) [inline scripts/launch_local_testnet.sh:269]: Can these commands fail and exit the script early?

File: scripts/launch_local_testnet.sh [hunk 4]
```diff
@@ -438,16 +468,16 @@ for NUM_NODE in $(seq 0 $(( NUM_NODES - 1 ))); do
     done
   fi
 
-  ./build/nimbus_beacon_node \
+  $BEACON_NODE_COMMAND \
     --non-interactive \
     --nat:extip:127.0.0.1 \
-    --network="${DATA_DIR}" \
+    --network="${CONTAINER_DATA_DIR}" \
```
Comment (stefantalpalaru) [inline scripts/launch_local_testnet.sh:474]: We end up using "CONTAINER_" variables even when not using a container?

**Review by stefantalpalaru — APPROVED** [review]

File: scripts/launch_local_testnet.sh (hunk 4 above)
Comment (zah) [reply scripts/launch_local_testnet.sh:474]: Yes, then `CONTAINER_DATA_DIR` is just an alias for `DATA_DIR`.

File: scripts/launch_local_testnet.sh (hunk 3 above)
Comment (zah) [reply scripts/launch_local_testnet.sh:269]: These commands fail from time to time for me actually (when the containers were already stopped). They produce benign error messages though and I didn't try to search for alternative solutions.

File: scripts/launch_local_testnet.sh (hunk 2 above)
Comment (zah) [reply scripts/launch_local_testnet.sh:242]: Doesn't hurt to be explicit

**Pull request merged**